JenkinsFile

```diff
@@ -0,0 +1,129 @@
+pipeline {
+    agent {
+        dockerfile {
+            label 'linux'
+        }
+    }
+    environment {
+        GIT_CREDENTIALS = credentials("${GITHUB_CREDENTIALS}")
+    }
+    stages {
+        stage('Lint/Test') {
+            steps {
+                sh """
+                    npm i
+                    npm run lint
+                    npm run test
+                """
+            }
+        }
+
+        stage('Version Check') {
+            steps {
+                 script {
+                    def data = readJSON file:'package.json'
+                    checkVersionAlreadyExists(data.version)
+                }
+            }
+        }
+
+        stage('Publish') {
+            steps {
+                sh """
+                    npm run build
```

Comment:
Should we separate out the build step? Build is a distinct item separate from publish.

Reply:
I think in this case they're not as separate. "Build" in this case is really just "Transpile to Js"

Reply:
I guess I should change the script name to match this. It should be `npm run transpile`